code/modules/client/preferences_savefile.dm

```diff
@@ -141,50 +139,49 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 		var/datum/keybinding/conflicted = item
 		to_chat(parent, span_danger("[conflicted.category]: [conflicted.full_name] needs updating"))
 
-
-/datum/preferences/proc/load_path(ckey,filename="preferences.sav")
+/datum/preferences/proc/load_path(ckey, filename="preferences.json")
 	if(!ckey)
 		return
-	path = "data/player_saves/[ckey[1]]/[ckey]/[filename]"
+	// path = "data/player_saves/[ckey[1]]/[ckey]/[filename]"
+	path = "data/player_saves_json_testing/[ckey[1]]/[ckey]/[filename]"
 
-/datum/preferences/proc/load_preferences()
+/datum/preferences/proc/load_savefile()
 	if(!path)
-		return FALSE
-	if(!fexists(path))
-		return FALSE
+		CRASH("Attempted to load savefile without first loading a path!")
+	savefile = new /datum/json_savefile(path)
 
-	var/savefile/S = new /savefile(path)
-	if(!S)
+/datum/preferences/proc/load_preferences()
+	if(!savefile)
+		stack_trace("Attempted to load the preferences of [parent] without a savefile; did you forget to call load_savefile?")
+		load_savefile()
+	if(!savefile)
+		stack_trace("Failed to load the savefile for [parent] after manually calling load_savefile; something is very wrong.")
 		return FALSE
-	S.cd = "/"
 
-	var/needs_update = savefile_needs_update(S)
+	var/needs_update = save_data_needs_update(savefile.get_entry())
 	if(needs_update == -2) //fatal, can't load any data
 		var/bacpath = "[path].updatebac" //todo: if the savefile version is higher then the server, check the backup, and give the player a prompt to load the backup
 		if (fexists(bacpath))
 			fdel(bacpath) //only keep 1 version of backup
-		fcopy(S, bacpath) //byond helpfully lets you use a savefile for the first arg.
+		fcopy(savefile.path, bacpath) //byond helpfully lets you use a savefile for the first arg.
```

Comment:
Comment here doesn't make much sense in this context because we aren't passing the savefile anymore in the first arg for `fcopy`, it was changed to use the path for the JSON file.